src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -337,3 +324,95 @@ void mongoRegistrationGet
 
   oeP->fill(SccOk, "");
 }
+
+
+
+/* ****************************************************************************
+*
+* mongoRegistrationsGet - 
+*/
+void mongoRegistrationsGet
+(
+  std::vector<ngsiv2::Registration>*  regV,
+  const std::string&                  tenant,
+  const std::vector<std::string>&     servicePathV,
+  OrionError*                         oeP
+)
+{
+  bool         reqSemTaken = false;
+  std::string  err;
+  mongo::OID   oid;
+  StatusCode   sc;
+  int          limit  = 5000;  // FIXME: Pagination: limit, offset and count are a preparation for it - see mongoGetSubscriptions()
+  int          offset = 0;
+  long long    count;
+
+  reqSemTake(__FUNCTION__, "Mongo Get Registrations", SemReadOp, &reqSemTaken);
+
+  LM_T(LmtMongo, ("Mongo Get Registrations"));
+
+  std::auto_ptr<mongo::DBClientCursor>  cursor;
+  mongo::Query                          q;
+
+  // FIXME P4: Return error if more than one service path in servicePathV ?
```

Comment:
This is a bit tricky... I undesrtand you mean that add an `else` to the `if` clause so we return an OrionError in that case.

Looking to the parallel code in subscription:

```
  if (!servicePath.empty() && servicePath != "/#")
  {
    q = Query(BSON(CSUB_SERVICE_PATH << servicePath));
  }
  else
  {
    q = Query();
  }
```

in that case the `else` is used to do servicepath-unfiltered queries. In would be the way of achiving the "05. GET all four registrations, without service path" case in the functional test.

Not sure which is the best option (either returning error or doing servicepath-unfiltered queries). In the latter case we are providing more functionality, although the usage model is a bit crazy (as many things in general with regards to service path).

(Side-note: preferred way to check for string emptness is `.empty()`, see https://stackoverflow.com/questions/41425569/checking-for-empty-string-in-c-alternatives)

Reply:
Yes, I mean to do an else with error. Really, it should be done before, in the service routine.
A have a doubt about the
```
else
{
    q = Query();
}
```
My doubt it ... what does this change?
Before, `q` is declared as `Query q` and after that, `q` is assigned `Query()`, just to call the constructor again?  I may be missing something here ... seems like a waste of time to me ...

Apart from all this, as the functest with `/#` doesn't do what it should (thank you for noticing my lapsus), something has to be wrong ...

Reply:
Fixed in 725e934